skorch/classifier.py

```diff
@@ -99,6 +123,8 @@ def check_data(self, X, y):
                    "``iterator_train`` and ``iterator_valid`` parameters "
                    "respectively.")
             raise ValueError(msg)
+        if y is not None:
+            self.classes_inferred_ = np.unique(y)
```

Comment:
`classes_inferred_` can be private.

Reply:
That's the first thing I did, but this has unintended side effects on `clone`, resulting in it failing the consistency check. This might be something we should investigate further but I thought it won't hurt to have this attribute there. I wouldn't document it in case we do want to remove it later.